analyses/npl/09_flood_events_by_municipality.md

```diff
@@ -0,0 +1,300 @@
+    country_iso3=COUNTRY_ISO3)
+df_return_period =  pd.read_excel(GLOFAS_RP_FILENAME, index_col='rp')
+```
+
+Read in events and clean
+
+```python
+df_events = (pd.read_excel(PAST_EVENTS_FILENAME, index_col='S.No.')
+             .rename(columns={'Mun_ETHOS_Code': 'pcode'})
+            )
+# Drop events with no date and convert date column
+df_events = df_events.loc[df_events['Year'] > 0]
+# For days that are 0, assume they mean the middle of the month
+df_events['Day'] = np.where(df_events['Day'] == 0, 15, df_events['Day'])
+df_events['Incident Date'] = pd.to_datetime(df_events[['Year', 'Month', 'Day']])
+
+# Only select events in the time range of GloFAS
+df_events = df_events.loc[df_events['Incident Date'] > ds_glofas_reanalysis.time[0].data]
+```
+
+Merge event data with admin
+
+```python
+df_admin = (gpd.read_file(f'zip://{ADMIN_SHAPEFILE}!{ADMIN2_SHAPEFILE}')
+            .rename(columns={'ADM2_PCODE': 'pcode'}))
+
+df_events = pd.merge(df_admin, df_events, how='right', left_on='pcode', right_on='pcode')
+```
+
+Read in Basin and Watershed to get areas of interest
+
+```python
+df_basins = gpd.read_file(BASINS_SHAPEFILE)
+df_watershed = gpd.read_file(WATERSHED_SHAPEFILE)
+```
+
+```python
+df_basins['Karnali'] = 0
+idx = df_basins['Major_Basi'].isin([f'{basin} River Basin' for basin in KARNALI_BASINS])
+df_basins.loc[idx, 'Karnali']= 1
+karnali = df_basins[idx].dissolve(by='Karnali')
+koshi = df_watershed[df_watershed['WSH_NME'] == f'{KOSHI_WATERSHED} Watershed']
+```
+
+Plot event municipalities over the areas of interest
+
+```python
+fig, ax = plt.subplots()
+karnali.plot(ax=ax)
+koshi.plot(ax=ax)
+df_events.boundary.plot(ax=ax, color='r', lw=0.5)
+```
+
+Might be easier to just do a latitude cut. Latitude > 86 is Koshi, < 83 is Karnali.
+
+```python
+df_events['basin'] = np.nan
+df_events.loc[df_events['geometry'].bounds.minx > 86,'basin'] = 'Koshi'
+df_events.loc[df_events['geometry'].bounds.minx < 83,'basin'] = 'Karnali'
+df_events = df_events.dropna(subset=['basin'])
+```
+
+### Combine events with GloFAS
+
+```python
+# Take only the columns with impact and group by the basin and incident date and sum
+df_events_final = df_events[df_events.columns[24:43]].groupby(['basin', 'Incident Date']).sum()
+
+```
+
+```python
+# Since there are two many events, get only those that are high impact
+# Use a rolling sum
+rolling_sum_size = 10
+center = False
+event_thresh = 0
+impact_parameters =  ['Total Death', 
+                         'Affected Family', 
+                         'Private House Partially Damaged', 
+                         'Private House Fully Damaged']
+plot_results = False
+
+df_events_high_impact = pd.DataFrame(columns=['date', 'basin', 'impact_parameter'])
+
+for basin, station in STATIONS.items():
+    # Reindex the events to the reanalysis time
+    # and get the rolling sum of impact
+    df = df_events_final.loc[basin]
+    df = (df.reindex(ds_glofas_reanalysis.time.data, fill_value=0)
```

Comment:
Am not totally following here on the logic behind the rolling sum. If each row in ```df_events_final``` is a discrete flood event with non consecutive dates, then what's the axis that we're rolling across (so to speak...)?

Reply:
It's just a way of smoothing the data, particularly in the case where there are several small events over the course of a few days, these should be grouped into a larger single event. I hope that makes sense.